day_03/solution.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"utils/readfile"
+)
+
+const tree = "#"
+const openSquare = "."
+
+type slope struct {
+	right int
+	down  int
+}
+
+func main() {
+	text := readfile.GetFileContents("../day_03/data.txt")
+
+	slopeArrSolution1 := []slope{{3, 1}}
+	slopeArrSolution2 := []slope{{1, 1}, {3, 1}, {5, 1}, {7, 1}, {1, 2}}
+
+	fmt.Println("solution 1: ", getMultipliedTreesFromSlopes(slopeArrSolution1, text))
+	fmt.Println("solution 2: ", getMultipliedTreesFromSlopes(slopeArrSolution2, text))
+}
+
+func getMultipliedTreesFromSlopes(slopes []slope, inputMap []string) int {
+	numOfTrees := 1
+
+	for _, slopeItem := range slopes {
+		numOfTrees = numOfTrees * getTreesFromSlope(slopeItem, inputMap)
+	}
+
+	return numOfTrees
+}
+
+func getCharFromIndex(text string, index int) string {
+	return string(text[index : index+1])
```

Comment:
isn't this just overcomplicated, text[index] should be enough, so this function does not do much